### PR TITLE
Added function to handle None inputs in bcs

### PR DESCRIPTION
**Pull request by jcgraciosa**
Added a method to SolverBaseClass to convert BCs inputted by the user as None into sympy.oo for subsequent use by the solver. Also added some type checks and handlers and it should not break the current example scripts (based on some tests).

### CODE REVIEW

**Comment by lmoresi** [comment]
Is this a fix for #149 ?

**Comment by jcgraciosa** [comment]
Yes, it's a fix for #149. When declaring boundary conditions, the user should put in None if that corresponding component is unconstrained. 

Also, the fix does not break the current examples (using sympy.oo for no constraints), but might be good idea if I'm allowed to change them soon.

File: src/underworld3/cython/petsc_generic_snes_solvers.pyx [hunk 1]
```diff
@@ -153,6 +153,36 @@ class SolverBaseClass(uw_object):
         self._is_setup = False
 
         return
+    
+    def _handle_none_bcs(self, fn):
+        # converts bcs put as None to sympy.oo 
+
+        import numpy as np
+
+        if hasattr(fn, "__len__"):      # BC input is iterable (has __len__ attribute)
```
Comment (julesghub) [inline src/underworld3/cython/petsc_generic_snes_solvers.pyx]: Instead of python "duck" typing, let's just assume fn is list/tuple of length `u.dim` for now?

**Comment by lmoresi** [comment]
Checking - are you happy with this, @julesghub ? We should not leave the branch hanging forever.

**Comment by julesghub** [comment]
Good point @lmoresi.
I'll test this against the new `dev` and see if it's compatible. Should be.

**Comment by julesghub** [comment]
@jcgraciosa, please review my changes and we can discuss further if needed.

**Review by jcgraciosa — COMMENTED** [review]
I am fine with the changes. 
Thanks for improving on the original version.
